kahuna/public/js/components/gr-image-metadata/gr-image-metadata.js

```diff
@@ -13,153 +13,150 @@ export const module = angular.module('gr.imageMetadata', [
-        });
-
-        ctrl.updateMetadataField = function (field, value) {
-            return editsService.updateMetadataField(ctrl.image, field, value)
-                .then((updatedImage) => {
-                    if (updatedImage) {
-                        ctrl.image = updatedImage;
-                        updateAbilities(updatedImage);
-                        $rootScope.$emit(
-                          'track:event',
-                          'Metadata',
-                          'Edit',
-                          'Success',
-                          null,
-                          {
-                            field: field,
-                            value: value
-                          }
-                        );
-                    }
-                })
-                .catch(() => {
-                  $rootScope.$emit(
-                    'track:event',
-                    'Metadata',
-                    'Edit',
-                    'Failure',
-                    null,
-                    {
-                      field: field,
-                      value: value
-                    }
-                  );
-
-                    /*
-                     Save failed.
-
-                     Per the angular-xeditable docs, returning a string indicates an error and will
-                     not update the local model, nor will the form close (so the edit is not lost).
-                     Instead, a message is shown and the field keeps focus for user to edit again.
-
-                     http://vitalets.github.io/angular-xeditable/#onbeforesave
-                     */
-                    return 'failed to save (press esc to cancel)';
-                });
-        };
-
-        ctrl.removeImageFromCollection = (collection) => {
-            ctrl.removingCollection = collection;
-            collections.removeImageFromCollection(collection, ctrl.image)
-                .then(() => ctrl.removingCollection = false);
-        };
-
-        ctrl.displayLeases = () => {
-            return ctrl.userCanEdit || ctrl.image.leases > 0;
-        };
-
-        $scope.$on('$destroy', function() {
-            freeUpdateListener();
-        });
+    function isUsefulMetadata(metadataKey) {
+      return ignoredMetadata.indexOf(metadataKey) === -1;
+    }
+
+    function updateAbilities(image) {
+      editsService.canUserEdit(image).then(editable => {
+        ctrl.userCanEdit = editable;
+      });
     }
+
+    const freeUpdateHandler = (updatedImage) => {
+      console.log('image update recvd in gr-image-metadata', updatedImage);
+      ctrl.image = updatedImage;
+      ctrl.usageRights = imageService(ctrl.image).usageRights;
+      ctrl.metadata = updatedImage.data.metadata;
+      ctrl.setUsageCategory(ctrl.usageCategories, ctrl.usageRights.data.category);
+    };
+
+    const freeUpdateListener = $rootScope.$on('images-updated', (e, updatedImages) => updatedImages.map(updatedImage => freeUpdateHandler(updatedImage)));
```

Comment:
What is a 'free update'? Is there a better name? (I know it's not your name – but it might be an opportunity to make things clearer!)

Reply:
Agreed, this is terribly named.

Reply:
Without better understanding of the function and the context in which it's run, I'm going to assume that `free` is important to the name in some obscure way that will become clear with sufficient introspection- and leave it be.

Reply:
Oh, I think I understand – it's the result of assigning an event handler. Calling it presumably removes the handler.

Eh, this is probably lack of angular domain knowledge rather than a naming problem – although 'remove' might make things clearer. Guessing it's a convention throughout, so best leave it 👍

Reply:
That makes sense, I'm going to get rid of the `free` from the handlers, which should imply this (absolutely genius) angular api design

Reply:
👍